vllm/entrypoints/chat_utils.py

```diff
@@ -386,6 +411,57 @@ def _get_full_multimodal_text_prompt(placeholder_counts: Dict[str, int],
 _RefusalParser = partial(cast, ChatCompletionContentPartRefusalParam)
 MODEL_KEEP_MULTI_MODAL_CONTENT = {'mllama'}
 
+# Define a mapping from part types to their corresponding parsing functions.
+MM_PARSER_MAP: Dict[str, Callable[[ChatCompletionContentPartParam], str]] = {
+    "text": lambda part: _TextParser(part)["text"],
+    "image_url": lambda part: _ImageParser(part)["image_url"]["url"],
+    "audio_url": lambda part: _AudioParser(part)["audio_url"]["url"],
+    "refusal": lambda part: _RefusalParser(part)["refusal"],
+}
+
+
+def _parse_chat_message_content_mm_part(
+        part: ChatCompletionContentPartParam) -> Tuple[str, str]:
+    """
+    Parses a given multi modal content part based on its type.
+
+    Args:
+        part: A dict containing the content part, with a potential 'type' field.
+
+    Returns:
+        A tuple (part_type, content) where:
+        - part_type: Type of the part (e.g., 'text', 'image_url').
+        - content: Parsed content or an empty string if unsupported.
+
+    Raises:
+        ValueError: If the 'type' field is missing and no direct URL is found.
+    """
+    part_type = part.get("type", None)  # type: ignore
```

Comment:
Do not ignore type if it can be easily fixed. In this case, I guess the following change should fix the type error
```python
part_type: Optional[str] = part.get("type", None)
```
Plz also try to fix other types.

Reply:
sounds good, used proper way addressed this, no type ignore now.